### PR TITLE
fix(npm): support dist tags specified in npm package dependencies

### LINKED ISSUE #16321: Tagged versions in npm package dependencies cause error
See https://github.com/denoland/deno/issues/15753#issuecomment-1264180597

> ```
> $ deno --version
> deno 1.26.0 (release, x86_64-unknown-linux-gnu)
> v8 10.7.193.3
> typescript 4.8.3
> 
> ~ ──────────────────────────────────────────────────────────────────────────────────────────────
> $ deno run -A --unstable --node-modules-dir npm:live-server@1.2.2
> error: npm package: live-server@1.2.2
> 
> Caused by:
>     0: error parsing version requirement for dependency: opn@latest
>     1: Invalid npm version requirement 'latest'.
>     2: Unexpected character.
>          latest
>          ~
> ```

### PR DESCRIPTION
**Pull request by dsherret**
Closes #16321

### CODE REVIEW

File: cli/npm/semver/mod.rs [hunk 1]
```diff
@@ -505,7 +607,9 @@ mod tests {
 
   #[test]
   pub fn npm_version_req_ranges() {
-    let tester = NpmVersionReqTester::new(">= 2.1.2 < 3.0.0 || 5.x");
+    let tester = NpmVersionReqTester::new(
+      ">= 2.1.2 < 3.0.0 || 5.x || ignored-invalid-range || $#$%^#$^#$^%@#$%SDF|||",
```
Comment (bartlomieju) [inline cli/npm/semver/mod.rs:611]: 🙃

**Review by bartlomieju — COMMENTED** [review]
Suggest to add an integration test

**Comment by dsherret** [comment]
> Suggest to add an integration test

I don't think we need one because it's localized to this area. The npm integration tests are really slow.

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**